File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.java
import java.util.*;
import java.util.Map.Entry;
class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
   
        
        HashMap<Integer, Integer> tangerineMap = new HashMap<>();
        
        for(int x : tangerine) {
        	tangerineMap.put(x, tangerineMap.getOrDefault(x, 0) + 1);
        }
        
        
        List<Integer> tangerineList = new ArrayList<>();
        
        for(int x : tangerineMap.keySet()) {
        	tangerineList.add(tangerineMap.get(x));
        }
        
        tangerineList.sort(Collections.reverseOrder());
        
     
        int sum = 0;
        
        for(int x : tangerineList) {
        	sum += x;
        	answer++;
        	if(sum >= k) {
        		break;
        	}
        }
        
       
        
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let map = new Map();
    tangerine.forEach((v) => {
        map.set(v, (map.get(v) || 0) + 1);
    })
    let counts = Array.from(map.values()).sort((a, b) => b - a);
    for (var i = 0; k > 0; i++) k -= counts[i];
    
    return i;
}
